translate/helpers/js_helpers.py

```diff
@@ -30,15 +36,39 @@ def program_to_gast(node):
     return gast
 
 """
-converts a python ast BinOp and converts it to a readable string recursively 
+converts a python ast BinOp and converts it to a gast node
 """
-def binOp_to_str(bop):
+def binOp(bop):
+  gast = {"type" : "binOp"}
+  if bop.left.type != "BinaryExpression" and bop.right.type != "BinaryExpression":
+      gast["left"] = js_router.node_to_gast(bop.left)
+      gast["op"] = bop.operator
+      gast["right"] = js_router.node_to_gast(bop.right)
   if bop.left.type == "BinaryExpression":
-    s = binOp_to_str(bop.left) + bop.operator + bop.right.raw
-  else:
-    # base case: if left is just a number, operate on left wrt right
-    s = bop.left.raw + bop.operator + bop.right.raw
-  return s
+      gast["left"] = binOp(bop.left)
+      gast["op"] = bop.operator
+      gast["right"] = js_router.node_to_gast(bop.right)
+  if bop.right.type == "BinaryExpression":
+      gast["left"] = binOp(bop.left)
+      gast["op"] = bop.operator
+      gast["right"] = js_router.node_to_gast(bop.right)
+  return gast
+
+def boolOp(node):
+    gast = {"type": "boolOp"}
+    if node.left.type != "LogicalExpression" and node.right.type != "LogicalExpression":
+        gast["left"] = js_router.node_to_gast(node.left)
+        gast["op"] = node.operator
+        gast["right"] = js_router.node_to_gast(node.right)
+    elif node.left.type == "LogicalExpression":
+        gast["left"] = boolOp(node.left)
```

Comment:
Could the `boolOp` here be replaced with `node_to_gast` too? Then there would be no need to check for `"LogicalExpression"` in here too and it would get a lot simpler. Same thing above for `binOp`.

Reply:
Need to work on merging so I am adding a todo linking to this comment in the code.

Reply:
Just as another data point, this is why I love having high test coverage so much. You can try  stuff out that you think might work and the tests will make sure all the things you care about haven't changed.